File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int min = nums[0];
        int max_diff = -1;

        for(int i=0; i<nums.length; i++){
            if(nums[i]>min){
                int diff = nums[i] - min;
                if(max_diff<diff){
                    max_diff = diff;
                }
            }
            else{
                min = nums[i];
            }
        }
        return max_diff;
    }
}

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumDifference = function(nums) {
    let maxProfit=0;

    for(let i=0; i<nums.length; i++){
        for(let j=i+1; j<nums.length; j++){
            let findProfit= nums[j]-nums[i]
            if(findProfit > maxProfit){
                maxProfit = findProfit;
            }
        }
    }
    return maxProfit === 0 ? -1 : maxProfit;
};